API.md

```diff
@@ -0,0 +1,552 @@
+# API
+
+* [Static Functions](#static-functions)
+  * [`create`](#create)
+* [Instance Methods](#instance-methods)
+  * [`start`](#start)
+  * [`stop`](#stop)
+  * [`dial`](#dial)
+  * [`dialProtocol`](#dialProtocol)
+  * [`hangUp`](#hangUp)
+  * [`handle`](#handle)
+  * [`unhandle`](#unhandle)
+  * [`peerRouting.findPeer`](#peerRouting.findPeer)
+  * [`contentRouting.findProviders`](#contentRouting.findProviders)
+  * [`contentRouting.provide`](#contentRouting.provide)
+  * [`_dht.put`](#_dht.put)
+  * [`_dht.get`](#_dht.get)
+  * [`_dht.getMany`](#_dht.getMany)
+  * [`pubsub.getPeersSubscribed`](#pubsub.getPeersSubscribed)
+  * [`pubsub.getTopics`](#pubsub.getTopics)
+  * [`pubsub.publish`](#pubsub.publish)
+  * [`pubsub.subscribe`](#pubsub.subscribe)
+
+## Static Functions
+
+### create
+
+Creates an instance of Libp2p.
+
+`create(options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| options | `Object` | libp2p options |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Libp2p>` | Promise resolves with the Libp2p instance |
+
+#### Example
+
+```js
+const Libp2p = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = await Libp2p.create(options)
+```
+
+Note: It is important pointing out that with `create`, the `PeerInfo` option is not required and will be generated if it is not provided.
+
+As an alternative, it is possible to create a Libp2p instance with the constructor:
+
+#### Example
+
+```js
+const Libp2p = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = new Libp2p (options)
+```
+
+Required keys in the `options` object:
```

Comment:
You just said above that `peerInfo` was not required.

Reply:
`peerInfo` is not required for the `create`. With the alternative approach,i.e. the `constructor` it is required. This logic is prior do this refactor, if you want to discuss this, maybe we can open an issue for it?

Reply:
Ok that makes sense.